polymath/extensions/vector_ops.py

```
@@ -4,691 +4,710 @@
-            raise IndexError(f'axis is out of range ({-arg._rank},{arg._rank}) in '
-                             f'{type(arg)}.{op}: {i}')
+            obj.insert_derivs(new_derivs)
 
-        if selections[i]:
-            raise IndexError(f'duplicated axis in {type(arg)}.{op}: {axis_for_show}')
+        return obj
 
-        selections[i] = True
+    @staticmethod
+    @staticmethod
+    def norm(arg, axis=-1, *, classes=(), recursive=True):
```

Ruff (0.14.7)

Missing return type annotation for staticmethod `norm`

(ANN205)